projects/packages/forms/src/contact-form/class-contact-form.php

```diff
@@ -1545,23 +1545,27 @@ public function process_submission() {
 			wp_schedule_event( time() + 250, 'daily', 'grunion_scheduled_delete' );
 		}
 
-		if (
-			$is_spam !== true &&
-			/**
-			 * Filter to choose whether an email should be sent after each successful contact form submission.
-			 *
-			 * @module contact-form
-			 *
-			 * @since 2.6.0
-			 *
-			 * @param bool true Should an email be sent after a form submission. Default to true.
-			 * @param int $post_id Post ID.
-			 */
-			true === apply_filters( 'grunion_should_send_email', true, $post_id )
-		) {
-			self::wp_mail( $to, "{$spam}{$subject}", $message, $headers );
+		if ( $is_spam !== true ) {
+			// We first need to check if `sendEmailOnSubmission` prop is set and prioritize that.
+			$send_email_on_submission = $this->attributes['sendEmailOnSubmission'];
+			if ( isset( $send_email_on_submission ) ) {
+				if ( $send_email_on_submission ) {
+					self::wp_mail( $to, "{$spam}{$subject}", $message, $headers );
+				}
+			} elseif (
+				/**
+				 * Filter to choose whether an email should be sent after each successful contact form submission.
+				 *
+				 * @module contact-form
+				 *
+				 * @since 2.6.0
+				 *
+				 * @param bool true Should an email be sent after a form submission. Default to true.
+				 */
+				true === apply_filters( 'grunion_should_send_email', true ) ) {
```

Comment:
I guess we'd need to update the docs of: https://developer.jetpack.com/hooks/grunion_should_send_email/ , but I'm not sure where we would do that.

Reply:
I'll revert this change as it seems it will break existing usages even though the main reason for this param seems to be for possible internal usage/logging etc..

The filter seems to be a `catch all` one and can't rely on the newly created `feedback` entity created.

Reply:
We can organise getting docs updated 👍 We'll need docs updated for devs, and customers for Jetpack and Dotcom.